### ISSUE TITLE
Table of contents rendered for active pane, not the note it is for

### ISSUE THREAD

**Issue by tprotopopescu** [Aidurber/obsidian-plugin-dynamic-toc#53]
Since upgrading to obsidian 13.30 the plugin renders the table of contents for whichever note is active, not the note it is intended for. 

To reproduce: 

Add the table of contents to a note. Open another note in a new pane and switch to it. The table of contents will now show the headings from the active pane. Switch back to the note that has the toc and the headings switch back.

If you click on one of the links in the toc while the other document is active the toc reverts to its proper links (the ones from the intended document) and work as expected.

Tested in the help vault. 

Let me know if I can help debug.

**Comment by claremacrae** [comment]
Thank you for logging this. I cane here to log the same issue.

> If you click on one of the links in the toc while the other document is active the toc reverts to its proper links (the ones from the intended document) and work as expected.

I think that workaround only works if the doc with the toc block is not pinned.

If it is pinned, clicking on a link opens a copy of that document in a new pane, which I doubt is ever the intended behaviour.

**Comment by Aidurber** [comment]
This is a side-effect of the "fix" for #48. With Live Preview, it seems that code blocks share the same processor instance. I've put a few hours into this problem and I can't come up with a solution that fixes both this bug and #48. 

Instead of rolling back the fix for #48 whilst I figure this out, I'm going to let this bug stay active since it's the lesser of two. Sorry for the delays I'm struggling with this one 😅

**Comment by claremacrae** [comment]
Thanks for the update - much appreciated.

**Comment by claremacrae** [comment]
Just had a thought. Are you on the Obsidian Discord? 

I’m guessing other plugin authors may have been tripped up by this, and so whether it would be worth your describing the issue in the plugin-dev channel there?

**Comment by Aidurber** [comment]
@claremacrae Yes, I did ask about it and it was suggested to create a code mirror plugin instead. However I'm going to see if I can find a solution that doesn't require me to rewrite this plugin.

**Comment by claremacrae** [comment]
Understood - thanks.

**Comment by claremacrae** [comment]
Just to note that on iOS it’s not just other open notes that affect the TOC contents. When Obsidian launches, the TOC blocks flash repeatedly, seemingly as each note in the vault is cached.

**Comment by claremacrae** [comment]
Or maybe it’s as each note with a TOC a block is read?

**Comment by tprotopopescu** [comment]
Thanks for the update! In a way, this is kind of a handy bug, since I get an automatic outline of whatever note I'm working on, assuming there is a note with a toc visible :)

**Comment by errysreez** [comment]
Hello,

If I may suggest an approach :

As mentionned by @tprotopopescu , it can become a handy bug, so instead of fullty getting rid of it (which requires rewriting the plugin so far), you can :
- add an option to pin the table (it will be no longer "dynamic") via a new option in the code block
- add the source file for the toc (we may this way, add the TOC of a note to an other one

So we will get something like this:

```toc
style: bullet
...
pinned: True | False
source: myNote.md
```

**Comment by PiousAeneas** [comment]
I like the idea of an option to pin the table (maybe make it a feature request?), though I wonder if leaving this bug as the default (as implied by having a second “source” option) might make the plug-in more confusing since my sense is that the expected functionality is that the TOC _in_ a note will be the TOC _for_ that note.
